packages/cli/src/commands/worker.ts

```diff
@@ -498,7 +498,7 @@ export class Worker extends BaseCommand {
 		}
 
 		// Make sure that the process does not close
-		await new Promise(() => {});
+		if (!inTest) await new Promise(() => {});
```

Comment:
I take it there's no other choice but in general we should avoid sprinkling `inTest` checks in production code.

Reply:
This was "temporary" code that [Jan had to add](https://github.com/n8n-io/n8n/pull/5461), after I broke this in #5452 .
At some point I hope to fix this properly, until then we need this ugly check 😞